Assets/Scripts/GunSlingerScript.cs

```diff
@@ -213,5 +229,26 @@ private void OnTriggerExit(Collider other)
             isOnRange = false;
         }
     }
+
+    private void TakingCoverDestination()
+    {
+        agent.isStopped = false;
+        target = DetectClosestAlly();
+        Vector3 oppositeDirection = transform.position - ((target - transform.position).normalized * 5f);
```

Comment:
should add the opposite direction to the transform.position instead of substracting it, and the target - transform.position should be swapped since this vector is pointing towards the player

Reply:
Done